recipes/libnl/all/conanfile.py

```diff
@@ -17,61 +20,44 @@ class LibNlConan(ConanFile):
     default_options = {"fPIC": True, "shared": False}
     build_requires = ( "flex/2.6.4", "bison/3.7.6" )
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def validate(self):
         if self.settings.os != "Linux":
             raise ConanInvalidConfiguration("Libnl is only supported on Linux")
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        config_args = [
-            "--prefix={}".format(tools.unix_path(self.package_folder)),
-        ]
-        if self.options.shared:
-            config_args.extend(["--enable-shared=yes", "--enable-static=no"])
-        else:
-            config_args.extend(["--enable-shared=no", "--enable-static=yes"])
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-        self._autotools.configure(configure_dir=self._source_subfolder, args=config_args)
-        return self._autotools
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.generate()
 
     def build(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.install()
-        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```

Comment:
it seems conan.tools.files.rm doesn't behave the same way this method does? what's the replacement for it?